tests/rbd/test_performance_immutable_cache.py

```diff
@@ -0,0 +1,280 @@
+
+    Args:
+        output (str): The output string containing timing information.
+
+    Returns:
+        float: The time in seconds.
+    """
+    real_time_str = output.split("\n")[1].split("\t")[1]
+    minutes, seconds = map(float, real_time_str[:-1].split("m"))
+    return minutes * 60 + seconds
+
+
+def create_and_test_io(client, rbd, config, mount_path, io_type):
+    """
+    Perform I/O operations on an RBD image and return the time taken.
+
+    Args:
+        client (object): The client node object.
+        rbd (object): The RBD object.
+        config (dict): Configuration dictionary.
+        mount_path (str): The path where the image is mounted.
+        io_type (str): The type of I/O operation, e.g., "write" or "read".
+
+    Returns:
+        str: The time taken for the I/O operation.
+    """
+    io_config = {
+        "rbd_obj": rbd,
+        "client": client,
+        "size": config["fio"]["size"],
+        "do_not_create_image": True,
+        "config": {
+            "file_size": config["fio"]["size"],
+            "file_path": [f"{mount_path}/file"],
+            "get_time_taken": True,
+            "image_spec": [config["image_spec"]],
+            "operations": {
+                "fs": "ext4",
+                "io": True,
+                "mount": True,
+                "map": True,
+            },
+            "skip_mkfs": False,
+            "cmd_timeout": 2400,
+            "io_type": io_type,
+        },
+    }
+    krbd_io_handler(**io_config)
+    return io_config["config"].get("time_taken", "")
+
+
+def perform_clone_io(client, rbd_obj, pool, image, mount_path, **kw):
+    """
+    Perform I/O operations on cloned images and return the time taken.
+
+    Args:
+        client (object): Client node object.
+        rbd (object): RBD object.
+        config (dict): Configuration dictionary.
+        pool (str): Pool name.
+        image (str): Image name.
+        mount_path (str): image to mount.
+
+    Returns:
+        str: Time taken for the I/O operation.
+    """
+    config = kw.get("config")
+    rbd = rbd_obj.get("rbd")
+
+    # Write Data to the Parent Image
+    fio_args = {
+        "client_node": client,
+        "pool_name": pool,
+        "image_name": image,
+        "io_type": "write",
+        "size": config["fio"]["size"],
+    }
+
+    run_fio(**fio_args)
+
+    # Create snapshot, protect, and clone for the image
+    snap = "image_snap" + random_string(len=5)
+    clone = "image_clone" + random_string(len=5)
+    snap_spec = f"{pool}/{image}@{snap}"
+    clone_spec = f"{pool}/{clone}"
+
+    create_snap_and_clone(rbd, snap_spec, clone_spec)
+
+    config["image_spec"] = clone_spec
+
+    #  First read of cloned image
+    first_read = create_and_test_io(client, rbd, config, mount_path, "read")
+
+    #  Second read of cloned image
+    second_read = create_and_test_io(client, rbd, config, mount_path, "read")
+
+    # Extract the 'real' time from the output
```

Comment:
What do you mean by "real" time?

Reply:
```real    0m7.671s
user    0m0.281s
sys     0m0.686s
```
actual time took for fio run when we initiate fio with time option. extracting that time.